### PR TITLE
[ucal] Implement get_tz_data_version()

### PR DESCRIPTION
**Pull request by kpozin**

### CODE REVIEW

File: rust_icu_ucal/src/lib.rs [hunk 1]
```diff
@@ -136,6 +136,21 @@ pub fn get_default_time_zone() -> Result<String, common::Error> {
     String::try_from(&uchar)
 }
 
+/// Implements `ucal_getTZDataVersion`
+pub fn get_tz_data_version() -> Result<String, common::Error> {
+    let mut status = common::Error::OK_CODE;
+
+    let tz_data_version = unsafe {
+        let raw_cstring = versioned_function!(ucal_getTZDataVersion)(&mut status);
+        assert!(common::Error::is_ok(status));
```
Comment (filmil) [inline rust_icu_ucal/src/lib.rs]: Any special reason that you prefer an `assert!` here to returning the error?

Comment (kpozin) [reply rust_icu_ucal/src/lib.rs]: I was following the pattern used in `set_default_time_zone` and `get_default_time_zone` above. I can abandon the pattern here.

Comment (filmil) [reply rust_icu_ucal/src/lib.rs]: Yes please. Likely, it's the other functions that need to be corrected to return the error instead.

**Comment by filmil** [comment]
LGTM, would you mind committing once checks are done. Thx.

**Pull request merged**